### PR TITLE
Remove most calls to localize.

### PR DESCRIPTION
**Pull request by drwells**
We still use it in IBFESurfaceMethod which is in the process of being rewritten anyway.

@jabrown893 Please make similar fixes to the TAVR code, should that use `localize` in a similar way anywhere.

Related to #657.

### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

File: examples/IBFE/explicit/ex0/example.cpp [hunk 1]
```diff
@@ -531,7 +531,8 @@ main(int argc, char* argv[])
             System& X_system = equation_systems->get_system<System>(IBFEMethod::COORDS_SYSTEM_NAME);
             NumericVector<double>* X_vec = X_system.solution.get();
             NumericVector<double>* X_ghost_vec = X_system.current_local_solution.get();
-            X_vec->localize(*X_ghost_vec);
+            *X_ghost_vec = *X_vec;
```
Comment (boyceg) [inline examples/IBFE/explicit/ex0/example.cpp:534]: cannot check right now, but is this equivalent to what `copy_and_synch` does?

Comment (drwells) [reply examples/IBFE/explicit/ex0/example.cpp:534]: It does! Should we just use that instead?

Comment (boyceg) [reply examples/IBFE/explicit/ex0/example.cpp:534]: Yes?

**Comment by drwells** [comment]
Good point. Fixed!

**Comment by jabrown893** [comment]
Yes, the TAVR code called this function four times.

**Pull request merged**